cranelift/codegen/src/isa/aarch64/lower_inst.rs

```diff
@@ -2056,6 +2056,29 @@ pub(crate) fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
             }
         }
 
+        Opcode::ScalarToVector => {
+            let rn = put_input_in_reg(ctx, inputs[0], NarrowValueMode::None);
+            let rd = get_output_reg(ctx, outputs[0]);
+            let input_ty = ctx.input_ty(insn, 0);
+            let size = VectorSize::from_ty(ty.unwrap());
+            if (input_ty == I32 && ty.unwrap() == I32X4)
+                || (input_ty == I64 && ty.unwrap() == I64X2)
+            {
+                lower_constant_f128(ctx, rd, 0);
+                ctx.emit(Inst::MovToVec {
```

Comment:
This should simply be `Inst::MovToFpu`, and then there is no need for `lower_constant_f128()`.

BTW if the original load is a FP load, then this could be a move.

Reply:
From reading of `FMOV (general)`, I don't see anything that implies that lanes 1 and above of the destination register are zeroed.  I may well have missed it though; can you clarify?

Reply:
The [pseudocode](https://developer.arm.com/docs/ddi0596/latest/simd-and-floating-point-instructions-alphabetic-order/fmov-general-floating-point-move-to-or-from-general-purpose-register-without-conversion) for `FMOV (general)` uses the [assignment form](https://developer.arm.com/docs/ddi0596/latest/shared-pseudocode-functions/aarch64-functionsregisters-pseudocode#impl-aarch64.V.write.1) of `Vpart[]`, which zero-extends the written value if `part` is `0`, as the comments in the pseudocode state.

Reply:
I rewrote it to generate `FMOV` only.